src/Microsoft.AspNetCore.Mvc.Core/Formatters/DefaultMediaTypeRegistry.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Mvc.Formatters
+{
+    /// <summary>
+    /// A default implementation of <see cref="MediaTypeRegistry"/>.
+    /// </summary>
+    public class DefaultMediaTypeRegistry : MediaTypeRegistry
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="DefaultMediaTypeRegistry"/>.
+        /// </summary>
+        public DefaultMediaTypeRegistry()
```

Comment:
This could go a lot further. Ideally we'd support `[Produces]` on model types, and make this part of api explorer AND formatters. 

This kind of thing pairs well with media types like `application/product+json`